contracts/index/DepositZap.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: BUSDL-1.1
+pragma solidity 0.6.11;
+pragma experimental ABIEncoderV2;
+
+import {IDetailedERC20} from "contracts/common/Imports.sol";
+import {SafeERC20, SafeMath} from "contracts/libraries/Imports.sol";
+
+import {IERC4626} from "./Imports.sol";
+import {ICurve3Pool} from "./ICurve3Pool.sol";
+
+/**
+ * @notice Helper contract to swap from DAI/USDC/Tether to 3Pool token
+ * and then deposit for index tokens.
+ */
+contract DepositZap {
+    using SafeMath for uint256;
+    using SafeERC20 for IDetailedERC20;
+
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address public constant USDT = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
+
+    address public constant CURVE_3POOL =
+        0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7;
+    address public constant CURVE_3CRV =
+        0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490;
+
+    address public immutable indexToken;
+
+    mapping(uint256 => address) public underlyers;
+
+    constructor(address indexToken_) public {
+        indexToken = indexToken_;
+
+        underlyers[0] = DAI;
+        underlyers[1] = USDC;
+        underlyers[2] = USDT;
+    }
+
+    /**
+     * @notice deposit 3Pool underlyer amount for index shares
+     * @dev user must approve deposit zap for underlyer to transfer
+     * @param amount the amount of 3Pool underlyer
+     * @param index the index specifying 3Pool underlyer
+     */
+    function deposit(uint256 amount, uint8 index) external {
+        require(index < 3, "INVALID_INDEX");
+
+        // user must have approved zap for 3Pool underlyer
+        _addLiquidityOneCoin(amount, index, 0);
```

Comment:
What are your thoughts on making the `minAmount` zero here?

My thought is to have that as a parameter for `deposit` so the UI can set appropriate slippage. The UI already has slippage form controls used for the trade order type.